File: core/src/net/natruid/jungle/utils/ai/conditions/IsOverrideBehaviorCondition.java
package net.natruid.jungle.utils.ai.conditions;

import net.natruid.jungle.utils.ai.BehaviorCondition;

public class IsOverrideBehaviorCondition extends BehaviorCondition {
    private final String behaviorName;

    public IsOverrideBehaviorCondition(String behaviorName) {
        super(false);
        this.behaviorName = behaviorName;
    }

    @Override
    public void reset() {
    }

    @Override
    public boolean run() {
        return mBehavior.get(getSelf()).tree.overrideBehavior.equals(behaviorName);
    }
}

File: core/src/net/natruid/jungle/utils/ai/conditions/IsOverrideBehaviorCondition.kt
package net.natruid.jungle.utils.ai.conditions

import net.natruid.jungle.utils.ai.BehaviorCondition

class IsOverrideBehaviorCondition(private val behaviorName: String) : BehaviorCondition(false) {
    override fun reset() {
    }

    override fun run(): Boolean {
        return mBehavior[self].tree.overrideBehavior == behaviorName
    }
}
